tools/code-generation/generator/src/main/resources/com/amazonaws/util/awsclientgenerator/velocity/cpp/RequestEventStreamHandlerHeader.vm

```diff
@@ -38,6 +39,7 @@ namespace Model
 
     class ${operation.name}Handler : public Aws::Utils::Event::EventStreamHandler
     {
+        typedef std::function<void(const ${operation.name}InitialResponse&)> ${operation.name}ConnectionResponseCallback;
```

Comment:
its returning the same structure as initial response? should we just merge the 2 callbacks? do users ever need to know the difference?

Reply:
technically it is still 2 replies.
I've just re-used the same structure for convenience (of us and the user).

ConnectionResponse is an equivalent of java sdk event stream operation level response such as [StartStreamTranscriptionResponse](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/transcribestreaming/model/StartStreamTranscriptionResponse.html). Ideally, we should refactor our event stream API to provide "operation level response too" (instead of NoResult), but it is backward incompatible. Additionally, our NoResult is presented to the user callback AFTER the streaming has completed, while java sdk Operation Response is presented at the beginning of streaming session.
Technically this is not a event stream streaming event.

while initial-response is a service event

While they technically serve the same purpose, the underlying mechanism is quite different and we don't know at the codegen time which one will be actually used by a AWS Service.
(I guess all will reply on HTTP level and in addition initial-response can be also sent).

Reply:
I understand the difference in whats going on in background, it just seems pretty counter intuitive from users perspective to separate them - on some apis you have to check one callback and on other apis you check a different one. This is just a rant and maybe an oversight with original design of event streams. From user perspective i would want to receive initial response data once without carrying whether its part of http response or event stream event.

Anyways, can we at least doc in more detail? i guess from our perspective we would not know which api uses which, but we can be explicit in handler that you should expect both and verify doc for which one is used.

Reply:
Updated to use single callback for both. This callback may be called more than once though, depending on a server behavior.
Docstring is updated too to reflect this.